backend/loi-service/src/main/java/org/egov/works/repository/rowmapper/LOIRowMapper.java

```diff
@@ -0,0 +1,83 @@
+package org.egov.works.repository.rowmapper;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import digit.models.coremodels.AuditDetails;
+import org.egov.works.web.models.LetterOfIndent;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.jdbc.core.ResultSetExtractor;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.*;
+
+@Repository
+public class LOIRowMapper implements ResultSetExtractor<List<LetterOfIndent>> {
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @Override
+    public List<LetterOfIndent> extractData(ResultSet rs) throws SQLException, DataAccessException {
+
+        Map<String, LetterOfIndent> loiMap = new LinkedHashMap<>();
+        while (rs.next()) {
+            String id = rs.getString("id");
+            String tenantId = rs.getString("tenantId");
+            String loiNumber = rs.getString("loi_number");
+            String workPkgNumber = rs.getString("work_pkg_number");
+            String workIdentificationNumber = rs.getString("work_identification_number");
+            String fileNumber = rs.getString("file_number");
+            BigDecimal negotiatedPercentage = rs.getBigDecimal("negotiated_percentage");
+            BigDecimal agreementDate = rs.getBigDecimal("agreement_date");
+            String contractorId = rs.getString("contractor_id");
+            BigDecimal securityDeposit = rs.getBigDecimal("security_deposit");
+            String bankGaurante = rs.getString("bank_gaurante");
```

Comment:
I have renamed the attribute in the db as the first one had a typo.

Reply:
Fixed.